test/lighthouse/lighthouse-e2e.test.ts

```diff
@@ -38,129 +41,158 @@ const FLOW_JSON_REGEX = /window\.__LIGHTHOUSE_FLOW_JSON__ = (.*);<\/script>/;
 
 const execFileAsync = promisify(execFile);
 
-describe('Lighthouse stringify Puppeteer script', function () {
+export async function generateFlowResultViaStringify(
+  flow: UserFlow
+): Promise<FlowResult> {
+  fs.mkdirSync(TMP_DIR, { recursive: true });
+  const testTmpDir = fs.mkdtempSync(`${TMP_DIR}/lighthouse-`);
+  const scriptPath = `${testTmpDir}/stringified.cjs`;
+
+  const scriptContents = await stringify(flow, {
+    extension: new LighthouseStringifyExtension(),
+  });
+
+  snapshot(scriptContents);
+  fs.writeFileSync(scriptPath, scriptContents);
+
+  const { stdout, stderr } = await execFileAsync('node', [scriptPath], {
+    timeout: 50_000,
+  });
+
+  // Ensure script didn't quietly report an issue.
+  assert.strictEqual(stdout, '');
+  assert.strictEqual(stderr, '');
+
+  const reportHtml = fs.readFileSync(`${testTmpDir}/flow.report.html`, 'utf-8');
+  const flowResultJson = FLOW_JSON_REGEX.exec(reportHtml)?.[1];
+  if (!flowResultJson) throw new Error('Could not find flow json');
+
+  fs.rmSync(testTmpDir, { recursive: true, force: true });
+
+  return JSON.parse(flowResultJson);
+}
+
+async function generateFlowResultViaRunner(
+  flow: UserFlow
+): Promise<FlowResult> {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  const lighthouseExtension = new LighthouseRunnerExtension(browser, page);
+  const runner = await createRunner(flow, lighthouseExtension);
+  const result = await runner.run();
+
+  await page.close();
+  await browser.close();
+
+  assert.ok(result);
+
+  return lighthouseExtension.createFlowResult();
```

Comment:
We need to get some secondary result from the extension and this method made the most sense to me.

Another option is to add an `onFlowResultCreated` callback to the extension constructor or something.